scale-decode/src/lib.rs

```diff
@@ -174,6 +174,20 @@ pub trait DecodeAsType: Sized {
         input: &mut &[u8],
         type_id: u32,
         types: &PortableRegistry,
+    ) -> Result<Self, Error> {
+        Self::decode_as_type_maybe_compact(input, type_id, types, false)
+    }
+
+    /// Given some input bytes, a `type_id`, and type registry, attempt to decode said bytes into
+    /// `Self`. Implementations should modify the `&mut` reference to the bytes such that any bytes
+    /// not used in the course of decoding are still pointed to after decoding is complete.
+    ///
+    /// If is_compact=true, it is assumed the value is compact encoded (only works for some types).
+    fn decode_as_type_maybe_compact(
```

Comment:
let's add `#[doc(hidden)]` to this, because I hope that it's only really needed for internal use, and it'd be nice if people only saw/used `decode_as_type` :)